drivers/stepper_control/zephyr_stepper_motion_control/stepper_motion_control_timing_source.h

```diff
@@ -7,32 +7,55 @@
 #define ZEPHYR_DRIVER_STEPPER_STEP_DIR_STEPPER_TIMING_SOURCE_H_
 
 #include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/counter.h>
+
+struct timing_source_config {
+	const struct device *counter;
+};
+
+struct timing_source_data {
+	const struct device *motion_control_dev;
+	uint64_t microstep_interval_ns;
+	struct k_work_delayable stepper_dwork;
+
+#ifdef CONFIG_ZEPHYR_STEPPER_MOTION_CONTROL_COUNTER_TIMING
+	struct counter_top_cfg counter_top_cfg;
+	bool counter_running;
```

Comment:
counter_running is not needed because `is_running` api function does not seem to be used. also i don't know why we would want to use it from motion controller since motion controller is aware of the current running status itself.